include/mbedtls/ecp.h

```diff
@@ -227,450 +250,501 @@ mbedtls_ecp_keypair;
                      void *p_rng );
 
 /**
- * \brief           Generate a keypair
+ * \brief           This function generates an ECP keypair.
  *
- * \param grp       ECP group
- * \param d         Destination MPI (secret part)
- * \param Q         Destination point (public part)
- * \param f_rng     RNG function
- * \param p_rng     RNG parameter
+ * \note            This function uses bare components rather than an 
+ *                  mbedtls_ecp_keypair() structure to ease use with other 
+ *                  structures such as mbedtls_ecdh_context() or 
+ *                  mbedtls_ecdsa_context().
  *
- * \return          0 if successful,
- *                  or a MBEDTLS_ERR_ECP_XXX or MBEDTLS_MPI_XXX error code
+ * \param grp       The ECP group.
+ * \param d         The destination MPI (secret part).
+ * \param Q         The destination point (public part).
+ * \param f_rng     The RNG function.
+ * \param p_rng     The RNG context.
  *
- * \note            Uses bare components rather than an mbedtls_ecp_keypair structure
- *                  in order to ease use with other structures such as
- *                  mbedtls_ecdh_context of mbedtls_ecdsa_context.
+ * \return          \c 0 on success.
+ * \return          An \c MBEDTLS_ERR_ECP_XXX or \c MBEDTLS_MPI_XXX error code
+ *                  on failure.
  */
 int mbedtls_ecp_gen_keypair( mbedtls_ecp_group *grp, mbedtls_mpi *d, mbedtls_ecp_point *Q,
                      int (*f_rng)(void *, unsigned char *, size_t),
                      void *p_rng );
 
 /**
- * \brief           Generate a keypair
+ * \brief           This function generates an ECP key.
  *
- * \param grp_id    ECP group identifier
- * \param key       Destination keypair
- * \param f_rng     RNG function
- * \param p_rng     RNG parameter
+ * \param grp_id    The ECP group identifier.
+ * \param key       The destination key.
+ * \param f_rng     The RNG function.
+ * \param p_rng     The RNG context.
  *
- * \return          0 if successful,
- *                  or a MBEDTLS_ERR_ECP_XXX or MBEDTLS_MPI_XXX error code
+ * \return          \c 0 on success.
+ * \return          An \c MBEDTLS_ERR_ECP_XXX or \c MBEDTLS_MPI_XXX error code
+ *                  on failure.
  */
 int mbedtls_ecp_gen_key( mbedtls_ecp_group_id grp_id, mbedtls_ecp_keypair *key,
                 int (*f_rng)(void *, unsigned char *, size_t), void *p_rng );
 
 /**
- * \brief           Check a public-private key pair
+ * \brief           This function checks a public-private key pair.
+ *
+ * \param pub       The keypair structure holding the public key.
+ * \param prv       The keypair structure holding the private key.
  *
- * \param pub       Keypair structure holding a public key
- * \param prv       Keypair structure holding a private (plus public) key
+ * \note            The both are keypairs, and may optionally hold the corresponding other key, but the public key passed in thee pub is checked against the private key passed in prv.
```

Comment:
I find the wording of this sentence hard to follow. It is not immediately obvious what “the both” (why not “both”?) refers to, or what “the other key” means. “Thee” needs to be removed. “But” is not really appropriate because the opposition is not between the first and second part of the sentence, but between the first part and what is missing from the second part. Furthermore the first part is misleading: `prv` must contain the public part, the only thing that's optional is `pub` containing a private part.

Rather than having this note, I suggest adding a clarification to the parameter descriptions and to the function description.

> This function checks that the keypair objects `\p pub` and `\p prv` have the same group and the same public point, and that the private key in `\p prv` is consistent with the public key.

> `\param pub` The keypair structure holding the public key. If it contains a private key, that part is ignored.
> `\param prv` The keypair structure holding the a full key pair.

Reply:
much clearer, thank you. Done.